skbio/metadata/intersection.pyx

```diff
@@ -0,0 +1,513 @@
+"""
```

Comment:
@mortonjt @ebolyen see my comment [here](https://github.com/biocore/scikit-bio/pull/1298#issuecomment-207142069) re: including `IntervalTree` and custom index objects